src/cloud_provider/utilities.rs

```diff
@@ -218,22 +220,47 @@ pub fn generate_supported_version(
 
 // unfortunately some proposed versions are not SemVer like Elasticache (6.x)
 // this is why we need ot have our own structure
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
 pub struct VersionsNumber {
     pub(crate) major: String,
     pub(crate) minor: Option<String>,
     pub(crate) patch: Option<String>,
 }
 
+impl fmt::Display for VersionsNumber {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let mut version = vec![];
+        version.push(format!("{}", &self.major));
```

Comment:
you can replace with `let mut version = vec![self.major.clone()];`

Reply:
Plus, `version.push(format!("{}", &self.major));` should be `version.push(self.major.clone());`

Reply:
Why is it better to do version.push(self.major.clone()); ?

Reply:
because the format is unnecessary here. `format!(..)` is good to use when you want to concatenate strings/literals. What you want here is simply put your `self.major` string into a vec. You only need to clone it and put it directly into it.

Reply:
By the way, I just seen that Clippy suggest to use &self.major.to_string()

Reply:
`self.major.clone()` is good as well ^^